tests/code/test_minor.py

```diff
@@ -132,26 +131,225 @@ def test_getCommunityEngagementByTerm():
+                                "name": 'Habitat For Humanity',
+                                'term': 2,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData4 = {"id": 4,
+                                "matched": False, 
+                                "name": 'People Who Care',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData5 = {"id": 1,
+                                "matched": False, 
+                                "name": 'Databses',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'khatts'}
+        
+        neillzEngagementData1 = {"id": 4,
+                                "matched": False, 
+                                "name": 'People Who Care',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'neillz'}
+        
+        neillzEngagementData2 = {"id": 1,
+                                "matched": False, 
+                                "name": 'Databses',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'neillz'}
+        
+        setCommunityEngagementForUser('add', khattsEngagementData1, 'ramsayb2')
+        
+        khattsEngagements = list(IndividualRequirement.select())
+        assert khattsEngagements[0].course == Course.get_by_id(2)
+        assert khattsEngagements[0].program == None
+
+        # add 5 engagements and make sure the 5th raises the expected exception 
+        setCommunityEngagementForUser('add', khattsEngagementData2, 'ramsayb2')
+        setCommunityEngagementForUser('add', khattsEngagementData3, 'ramsayb2')
+        setCommunityEngagementForUser('add', khattsEngagementData4, 'ramsayb2')
+
+        with pytest.raises(DoesNotExist):
+            setCommunityEngagementForUser('add', khattsEngagementData5, 'ramsayb2')
+
+
+        # add records for another student and make sure it is added correctly. 
+        setCommunityEngagementForUser('add', neillzEngagementData1, 'ramsayb2')
+        neillzEngagements = list(IndividualRequirement.select())
+        assert neillzEngagements[3].username_id == 'khatts'
+        assert neillzEngagements[4].username_id == 'neillz'
+
+        # add a second record for that other student.
+        setCommunityEngagementForUser('add', neillzEngagementData1, 'ramsayb2')
+        neillz2Engagements = list(IndividualRequirement.select())
+        assert neillz2Engagements[3].username_id == 'khatts'
+        assert neillz2Engagements[4].username_id == 'neillz'
+        assert neillz2Engagements[5].username_id == 'neillz'
```

Comment:
Maybe not a huge issue, but I am not the biggest fan of the only tests being testing what is at each index. Feel like there could be some additional checks here.